pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -173,27 +215,105 @@ func (c *Controller) syncGKENetworkParamSet(ctx context.Context, key string) err
 		return nil
 	}
 
-	params := obj.(*networkv1.GKENetworkParamSet)
+	originalParams := obj.(*networkv1.GKENetworkParamSet)
+	params := originalParams.DeepCopy()
+
+	err = c.syncGNP(ctx, params)
+
+	if !reflect.DeepEqual(originalParams.Status, params.Status) {
+		if updateErr := c.updateGKENetworkParamSetStatus(ctx, params); updateErr != nil {
+			err = multierror.Append(updateErr, err)
+		}
+		if updateErr := c.updateGKENetworkParamSet(ctx, params); updateErr != nil {
+			err = multierror.Append(updateErr, err)
+		}
+	} else if !reflect.DeepEqual(originalParams, params) {
```

Comment:
another important thing on using DeepEqual is that it may lead to recomputation despite the objects are semantically the same, nil vs empty per example.

Kubernetes uses `apiequality.Semantic.DeepEqual` to avoid this problem instead of reflect.DeepEqual

https://pkg.go.dev/k8s.io/apimachinery/pkg/api/equality
https://github.com/search?q=repo%3Akubernetes%2Fkubernetes%20apiequality.Semantic.DeepEqual&type=code

Reply:
this can be a follow up